packages/block-library/src/navigation/editor.scss

```diff
@@ -3,7 +3,9 @@
  */
 
 // Undo default editor styles.
-// These need extra specificity.
+// These need extra specificity to do two things in classic themes
+// 1. Override the auto margin left from alignment styles.
+// 2. Override list style indentation.
 .editor-styles-wrapper .wp-block-navigation {
```

Comment:
If anything this needs to be a higher specificity, as the padding-left rule is not overriding the list styling applied by some older themes (e.g. Twenty Twenty). 😬

Reply:
Is this supposed to be overriding classic theme styles? How do this margin styles interact with the current block gap support for the Navigation block?

Reply:
The selectors target an inner element of the block, so block styles work ok. Block gap works ok as it's applied to the children of the `ul` being targeted in a nested rule.

[There's a separate bug in the nav block that the `ul` inner element has a `wp-block` classname](https://github.com/WordPress/gutenberg/pull/63801), so that's why the alignment styles need to be overridden.

I'll add some extra detail to the comment to make it clearer.

Reply:
In most cases I tested it was ok to wrap the `.editor-styles-wrapper` class in `:where()`. Unfortunately there were still some like TwentyTwentyOne which had selectors like `.wp-block-navigation [data-block]`. 

Keeping this selector's specificity as is, makes sense to me. Thanks for the clarifying comments.